cmd/utils.go

```diff
@@ -60,3 +64,73 @@ func scanForProjectFiles(projType projectType) ([]string, error) {
 	}
 	return paths, nil
 }
+
+// findProject scans the directory for Xcode Project (.xcworkspace / .xcodeproject) file first
+// If can't find any, ask the user to drag-and-drop the file
+func findXcodeProject() (string, error) {
+	var projpth string
+
+	projPaths, err := scanForProjectFiles(iOSProjectType)
+	if err != nil {
+		log.Printf("Failed: %s", err)
+		fmt.Println()
+
+		log.Infof("Provide the project file manually")
+		askText := `Please drag-and-drop your Xcode Project (` + colorstring.Green(".xcodeproj") + `) or Workspace (` + colorstring.Green(".xcworkspace") + `) file, 
+the one you usually open in Xcode, then hit Enter.
+(Note: if you have a Workspace file you should most likely use that)`
+		projpth, err = goinp.AskForPath(askText)
+		if err != nil {
+			return "", fmt.Errorf("failed to read input: %s", err)
+		}
+
+		return projpth, nil
+	}
+
+	if len(projPaths) == 1 {
+		log.Printf("Found one project file: %s.", path.Base(projPaths[0]))
+		return projPaths[0], nil
+	}
+
+	log.Printf("Found multiple project file: %s.", path.Base(projpth))
+	projpth, err = goinp.SelectFromStringsWithDefault("Select the project file you want to scan", 1, projPaths)
+	if err != nil {
+		return "", fmt.Errorf("failed to select project file: %s", err)
+	}
+
+	return projpth, nil
+}
+
+// findSolution scans the directory for Xamarin.Solution file first
+// If can't find any, ask the user to drag-and-drop the file
+func findXamarinSolution() (string, error) {
```

Comment:
This function does exactly the same as findXcodeProject. I think this function would be better to be renamed for a common one(findProjectFiles) with one input: projectType. Then the remaining logging can be refactored the way that it can be meant to the platform per projectType.